### PR TITLE
@co012/rpg 107 actions

### PR DESCRIPTION
**Pull request by co012**
# Description

# Checklist

* [x] Included code to test these changes
* [x] Updated Jira

### CODE REVIEW

File: src/main/java/io/rpg/controller/Controller.java [hunk 1]
```diff
@@ -171,15 +173,13 @@ public void onMouseClickedEvent(MouseClickedEvent event) {
...
-    if (distance < 1.5) {
-      if (object instanceof InteractiveGameObject) {
-        ((InteractiveGameObject) object).onAction();
-      }
 
-      if (object instanceof CollectibleGameObject) {
-        popupController.openTextImagePopup("Picked up an item!", objectView.getImage(), getWindowCenterX(), getWindowCenterY());
-        objectView.setVisible(false);
-        currentModel.removeGameObject(object);
+    if (distance < 1.5) {
```
Comment (ksiek127) [inline src/main/java/io/rpg/controller/Controller.java:177]: is the distance measured by squares? if yes, then 1.5 squares means i can pick up an item that is not adjacent to my square and this should not happen

**Review by ksiek127 — APPROVED** [review]
got a slight doubt about this 'distance < 1.5' in controller i pointed out in comment to controller.java, but the action logic itself looks good

File: src/main/java/io/rpg/controller/Controller.java (hunk 1 above)
Comment (co012) [reply src/main/java/io/rpg/controller/Controller.java:177]: I'll look into it

File: src/main/java/io/rpg/model/actions/Action.java [hunk 2]
```diff
@@ -4,4 +4,5 @@
  * A marker interface for action classes.
  */
 public interface Action {
+  Action VOID = new Action() {};
```
Comment (mhawryluk) [inline src/main/java/io/rpg/model/actions/Action.java:7]: Should other types of actions be initiated here or is that just for VOID?

Comment (co012) [reply src/main/java/io/rpg/model/actions/Action.java:7]: Only for VOID

**Pull request merged**